### PR TITLE
Fix: menubar throws errors on master focus trap too much recursion

### PR DESCRIPTION
**Pull request by vinicius73**
* Resolves: #2842 
* Target version: master 

### Summary

- use menu bar as container for emoji, upload and list actions (solves https://github.com/nextcloud/nextcloud-vue/issues/3154)
- fix emoji picker
  - add regression tests

### CODE REVIEW

**Review by juliushaertl — APPROVED** [review]

File: cypress/e2e/workspace.spec.js [hunk 1]
```diff
@@ -181,6 +181,17 @@ describe('Workspace', function() {
...
 
+	it('emoji picker', () => {
+		cy.openWorkspace()
+			.type('Let\'s smile together{enter}# ')
+
+		menuButton('emoji-picker').click()
+
+		cy.get('#emoji-mart-list button[aria-label="😀, grinning"]').first().click()
+
+		cy.getEditor().find('h1').should('contains', '😀')
```
Comment (juliushaertl) [inline cypress/e2e/workspace.spec.js:192]: Hm, seems to be failing on CI

**Comment by susnux** [comment]
For me it works now almost perfectly, with two issues remaining:

1. The image sub-menu still throws the focus trap recursion error
2. If I open the formatting help the focus is "trapped" somewhere else (I can not click the X to close it).

**Comment by vinicius73** [comment]
> 2\. If I open the formatting help the focus is "trapped" somewhere else (I can not click the X to close it).

I can't use the same solution in the modal 😢 
Modals can't contain other modals.

**Comment by susnux** [comment]
Thank you, the attachment sub-menu works perfectly now!
I am not sure if the formatting help issue is related, as it works, but the clickable area is limited to a small area (lower third)

@vinicius73 maybe you can also have a look at the table settings? As I get the same focus trap error when clicking on it and hover "delete this table".

**Review by susnux — APPROVED** [review]

**Comment by vinicius73** [comment]
> @vinicius73 maybe you can also have a look at the table settings? As I get the same focus trap error when clicking on it and hover "delete this table".

Sure, thanks to report it.

**Comment by vinicius73** [comment]
/backport 17c2da24267ba5e9edffc7bbe8abcc4bb29d0221, ded2e2c496beabe568938b3423f8e58e36a1ddb5, e2b2dee6b063a9a1f253d9e348e2679a6915a7f0 to stable25

**Comment by vinicius73** [comment]
/backport 17c2da24267ba5e9edffc7bbe8abcc4bb29d0221,ded2e2c496beabe568938b3423f8e58e36a1ddb5,e2b2dee6b063a9a1f253d9e348e2679a6915a7f0 to stable25

**Pull request merged**